### PR TITLE
Instance changes

### PR DESCRIPTION
**Pull request by tanvi029**

### CODE REVIEW

File: src/main/java/iudx/catalogue/server/authenticator/KcAuthenticationServiceImpl.java [hunk 1]
```diff
@@ -111,7 +111,8 @@ public AuthenticationService tokenInterospect(
   Future<Boolean> isValidEndpoint(String endpoint) {
     Promise<Boolean> promise = Promise.promise();
 
-    if (endpoint.equals(api.getRouteItems()) || endpoint.equals(api.getRouteInstance())) {
+    if (endpoint.equals(api.getRouteItems()) || endpoint.equals(api.getRouteInstance())
+            || endpoint.equals("/internal/ui/instance")) {
```
Comment (kailash) [inline src/main/java/iudx/catalogue/server/authenticator/KcAuthenticationServiceImpl.java:115]: Use an enum or constants file for String constants

**Review by kailash — CHANGES_REQUESTED** [review]
Use Future instead of callback handlers in service proxy definitions,

**Comment by kailash** [comment]
ok to test

File: src/main/java/iudx/catalogue/server/database/DatabaseServiceImpl.java [hunk 2]
```diff
@@ -1314,8 +1314,13 @@ public DatabaseService createMlayerInstance(
   }
 
   @Override
-  public DatabaseService getMlayerInstance(Handler<AsyncResult<JsonObject>> handler) {
-    String query = GET_MLAYER_INSTANCE_QUERY;
+  public DatabaseService getMlayerInstance(String id, Handler<AsyncResult<JsonObject>> handler) {
+    String query = "";
+    if (id.isBlank()) {
```
Comment (pranavrd) [inline src/main/java/iudx/catalogue/server/database/DatabaseServiceImpl.java:1319]: check for id == null before isBlank()

**Review by pranavrd — APPROVED** [review]
LGTM

**Review by kailash — APPROVED** [review]

**Pull request merged**